### PR TITLE
Enabled comparison ops for bfloat16 dtype on CPU

### PR DESCRIPTION
**Pull request by izdeby**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24224 [WIP] Enabled torch.mm and torch.mv for bfloat16
* #24183 Enabled masked methods for bfloat16
* **#24182 Enabled comparison ops for bfloat16 dtype on CPU**

-----
Fix: Enabled comparison operations for BFloat16 on CPU 
Test: via unit tests

Differential Revision: [D16763460](https://our.internmc.facebook.com/intern/diff/D16763460)

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -6915,52 +6915,65 @@ def reference(x, k, o3, o32):
...
-                torch.ne(x, b, out=boolRes)
-                self.assertEqual(byteRes.bool(), boolRes)
-
-                self.assertEquals(len(warningsCount), 6)
+            for dt in torch.testing.get_all_dtypes():
+                x = torch.tensor([1, 2, 3, 4], device=device, dtype=dt)
+                b = torch.tensor([2], device=device, dtype=dt)
+
+                if dt == torch.bool or dt == torch.half:
+                    continue
```
Comment (gchanan) [inline test/test_torch.py]: same as the the other review in the stack, stop doing this.

**Review by ezyang — APPROVED** [review]

**Review by gchanan — APPROVED** [review]
